src/render/SDL_render.c

```diff
@@ -915,22 +915,23 @@ static SDL_RenderLineMethod SDL_GetRenderLineMethod()
 
 static void SDL_CalculateSimulatedVSyncInterval(SDL_Renderer *renderer, SDL_Window *window)
 {
-    /* FIXME: SDL refresh rate API should return numerator/denominator */
-    int refresh_rate = 0;
+    double refresh_rate = 0.0;
     int display_index = SDL_GetWindowDisplayIndex(window);
     SDL_DisplayMode mode;
 
     if (display_index < 0) {
         display_index = 0;
     }
     if (SDL_GetDesktopDisplayMode(display_index, &mode) == 0) {
-        refresh_rate = mode.refresh_rate;
+        if (mode.refresh_rate.denominator) {
+            refresh_rate = (double) mode.refresh_rate.numerator / (double) mode.refresh_rate.denominator;
+        }
     }
     if (!refresh_rate) {
         /* Pick a good default refresh rate */
-        refresh_rate = 60;
+        refresh_rate = 60.0;
     }
-    renderer->simulate_vsync_interval_ns = (SDL_NS_PER_SECOND / refresh_rate);
+    renderer->simulate_vsync_interval_ns = (SDL_NS_PER_SECOND / (int)refresh_rate);
```

Comment:
You should use the numerator and denominator directly in this calculation rather than having a temporary double.

If you need me to, I can make that change after this lands.

Reply:
```
 918     int display_index = SDL_GetWindowDisplayIndex(window);
 919     SDL_DisplayMode mode;
 920     Uint32 num = 0;
 921     Uint32 den = 0;
 922 
 923     if (display_index < 0) {
 924         display_index = 0;
 925     }
 926     if (SDL_GetDesktopDisplayMode(display_index, &mode) == 0) {
 927         if (mode.refresh_rate.denominator) {
 928             num = mode.refresh_rate.numerator;
 929             den = mode.refresh_rate.denominator;
 930         }
 931     }
 932     if (num == 0) {
 933         /* Pick a good default refresh rate */
 934         num = 60;
 935         den = 1;
 936     }
 937     renderer->simulate_vsync_interval_ns = (SDL_NS_PER_SECOND * den) / num;
 938 }
```